library/Core/src/Juvix/Core/IR/CheckTerm.hs

```diff
@@ -208,6 +208,56 @@ typeTerm' term ann@(Typed.Annotation σ ty) =
       tAnn <- Typed.Annotation σ <$> substApp b s'
       t' <- typeTerm' t tAnn
       pure $ Typed.Pair s' t' ann
+    Core.CatProduct a b _ -> do
+      requireZero σ
+      void $ requireStar ty
+      a' <- typeTerm' a ann
+      b' <- typeTerm' b ann
+      pure $ Typed.CatProduct a' b' ann
+    Core.CatCoproduct a b _ -> do
+      requireZero σ
+      void $ requireStar ty
+      a' <- typeTerm' a ann
+      b' <- typeTerm' b ann
+      pure $ Typed.CatCoproduct a' b' ann
+    Core.CatProductIntro s t _ -> do
+      (π, a, b) <- requireCatProduct ty
+      let sAnn = Typed.Annotation (σ <.> π) a
+      let tAnn = Typed.Annotation (σ <.> π) b
+      s' <- typeTerm' s sAnn
+      t' <- typeTerm' t tAnn
+      pure $ Typed.CatProductIntro s' t' ann
+    Core.CatProductElimLeft a s _ -> do
+      a' <- typeTerm' a (Typed.Annotation σ (IR.VStar 0))
+      av <- evalTC a'
+      let sAnn = Typed.Annotation σ (IR.VCatProduct ty av)
+      s' <- typeTerm' s sAnn
+      pure $ Typed.CatProductElimLeft a' s' ann
+    Core.CatProductElimRight a s _ -> do
+      a' <- typeTerm' a (Typed.Annotation σ (IR.VStar 0))
+      av <- evalTC a'
+      let sAnn = Typed.Annotation σ (IR.VCatProduct av ty)
+      s' <- typeTerm' s sAnn
+      pure $ Typed.CatProductElimRight a' s' ann
+    Core.CatCoproductIntroLeft s _ -> do
+      (π, a, _b) <- requireCatCoproduct ty
+      let sAnn = Typed.Annotation (σ <.> π) a
+      s' <- typeTerm' s sAnn
+      pure $ Typed.CatCoproductIntroLeft s' ann
+    Core.CatCoproductIntroRight s _ -> do
+      (π, _a, b) <- requireCatCoproduct ty
+      let sAnn = Typed.Annotation (σ <.> π) b
+      s' <- typeTerm' s sAnn
+      pure $ Typed.CatCoproductIntroRight s' ann
+    Core.CatCoproductElim a b cp s t _ -> do
+      a' <- typeTerm' a (Typed.Annotation σ (IR.VStar 0))
```

Comment:
very minor, but we can have a const `(Typed.Annotation σ (IR.VStar 0))`

Reply:
Good idea.  Done.